src/components/buttons/flat-button/flat-button.js

```diff
@@ -1,48 +1,83 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
+import { css } from '@emotion/core';
+import vars from '../../../../materials/custom-properties';
 import withMouseOverState from '../../../hocs/with-mouse-over-state';
 import filterDataAttributes from '../../../utils/filter-data-attributes';
 import Text from '../../typography/text';
 import AccessibleButton from '../accessible-button';
-import styles from './flat-button.mod.css';
 
-export const FlatButton = props => {
-  const dataProps = {
-    'data-track-component': 'LinkButton',
-    ...filterDataAttributes(props),
-  };
+const getIconElement = props => {
+  if (!props.icon) return null;
 
   let iconTheme = 'black';
   if (props.isDisabled) iconTheme = 'grey';
   else if (props.tone === 'primary') iconTheme = 'green';
   else if (props.tone === 'secondary' && props.isMouseOver)
     iconTheme = 'orange';
 
-  const iconElement =
-    props.icon &&
-    React.cloneElement(props.icon, {
-      size: 'medium',
-      theme: iconTheme,
-    });
+  return React.cloneElement(props.icon, {
+    size: 'medium',
+    theme: iconTheme,
+  });
+};
+
+const getColorTone = (tone, isHover = false) => {
+  switch (tone) {
+    case 'primary':
+      return isHover ? vars['--color-green-25'] : vars['--color-green'];
+    case 'secondary':
+      return vars['--color-black'];
+    default:
+      return 'inherit';
+  }
+};
+
+export const FlatButton = props => {
+  const dataProps = {
+    'data-track-component': 'LinkButton',
+    ...filterDataAttributes(props),
+  };
 
   return (
     <div onMouseOver={props.handleMouseOver} onMouseOut={props.handleMouseOut}>
       <AccessibleButton
-        buttonAttributes={dataProps}
         type={props.type}
         label={props.label}
         onClick={props.onClick}
         isDisabled={props.isDisabled}
-        className={classnames(styles.button, styles[props.tone], {
-          [styles.disabled]: props.isDisabled,
-        })}
+        css={css`
+          display: flex;
+          align-items: center;
+          font-size: 1rem;
+          border: none;
+          background: none;
+          padding: 0;
+          min-height: initial;
+
+          > * + * {
+            margin: 0 0 0 ${vars['--spacing-4']};
+          }
+          p {
```

Comment:
This is because we are using Text.Body right? Maybe it makes more sense to allow Text.Body to accept a className and use `css` to pass the className for this.

Reply:
Yeah, I think you were trying to do the same right?

Reply:
I guess for now we leave it like it is and we make another PR to allow Text to accept classNames?

Reply:
Yep, makes sense